packages/apollo-angular/headers/src/index.ts

```diff
@@ -1,18 +1,24 @@
 import { HttpHeaders } from '@angular/common/http';
-import { ApolloLink, NextLink, Operation } from '@apollo/client/core';
+import { ApolloLink } from '@apollo/client/core';
 
-export const httpHeaders = () => {
-  return new ApolloLink((operation: Operation, forward: NextLink) => {
-    const { getContext, setContext } = operation;
-    const context = getContext();
+export class HttpHeadersLink extends ApolloLink {
```

Comment:
Apollo Client has moved toward classes for all links so I did the same here. I've kept the old function as deprecated, similar to what we've done in Apollo Client itself. I'll leave it up to you if you want to remove the old function completely or leave as-is.